gateways/c/fjage.c

```diff
@@ -287,13 +288,13 @@ static fjage_msg_t mqueue_get_any(fjage_gw_t gw, const char** clazzes, int clazz
   fjage_msg_t msg = NULL;
   if (clazzes == NULL || clazzlen < 1) return msg;
   for (int i = fgw->mqueue_tail; i != fgw->mqueue_head && msg == NULL; i++) {
+    if (i >= QUEUE_LEN) i = i%QUEUE_LEN;
     if (fgw->mqueue[i] != NULL) {
       const char* clazz1 = fjage_msg_get_clazz(fgw->mqueue[i]);
       if (clazz1 == NULL || ! mqueue_compare_any(clazz1, clazzes, clazzlen)) continue;
       msg = fgw->mqueue[i];
       fgw->mqueue[i] = NULL;
       if (i == fgw->mqueue_tail) fgw->mqueue_tail = (fgw->mqueue_tail+1) % QUEUE_LEN;
-      // break out
```

Comment:
Good to remove this, but any idea of the history of this comment, as it doesn't make sense here

Reply:
Looks like it was a comment added while development to point out the fact that if the code reached there it would automatically break out of the for-loop. I guess It was missed from being removed before PR.

https://github.com/org-arl/fjage/blame/b2b12fdb10df964c53414bdb798a29ccb9ce80b4/gateways/c/fjage.c#L296